File: _REPO/APRESS/distributed-.net-programming-in-vb-.net/Chapter08/SimpleComponent/SimpleCarLibrary/CarService.vb
'All the important enterprise service types are contained here.
Imports System.EnterpriseServices
Imports System.Runtime.InteropServices 'For GUID attribute
Imports System.Text 'For StringBuilder

 'Class requires JIT activation
<JustInTimeActivation(), _
 Guid("00E1160C-F65B-45fa-A92D-C24AB92831C8"), _
 EventTrackingEnabled()> _
Public Class CarService
   Inherits ServicedComponent
   
   ' Serviced components require a default constructor.
   Public Sub New()
   End Sub
   
   Public Function GetInfo() As String
      Dim ctxInfo As New StringBuilder()

      'Use ContextUtil to fetch context information
      ctxInfo.AppendFormat("Context ID:  {0}{1}", _
         ContextUtil.ContextId, vbCrLf)
      ctxInfo.AppendFormat("Activity ID: {0}{1}", _
         ContextUtil.ActivityId, vbCrLf)

      'If in transaction, get transaction ID
      Dim txId As String = "No Tx"
      If ContextUtil.IsInTransaction Then
         txId = ContextUtil.TransactionId.ToString()
      End If

      ctxInfo.AppendFormat("Transaction ID:    {0}{1}", txId, vbCrLf)
      ctxInfo.AppendFormat("Security Enabled?: {0}{1}", _
         ContextUtil.IsSecurityEnabled, vbCrLf)

      Return ctxInfo.ToString()
   End Function

End Class

File: _REPO/APRESS/distributed-.net-programming-in-csharp/Chapter08/SimpleComponent Server/SimpleCarLibrary/CarService.cs
using System;
using System.Diagnostics;              // event Logging
using System.EnterpriseServices;       // ServicedComponent
using System.Runtime.InteropServices;  // Class Interface
using System.Text;                     // StringBuilder
using System.Threading;

namespace SimpleCarLibrary
{
   [JustInTimeActivation(), 
    Guid("00E1160C-F65B-45fa-A92D-C24AB92831C8"),
    EventTrackingEnabled()]
   public class CarService : ServicedComponent
   {
      public CarService()
      {
         string logMsg = string.Format("Created Object: {0}", this.GetHashCode());
         EventLog.WriteEntry("CarServiced.ctor()", logMsg);
      }

      public string GetInfo()
      {
         StringBuilder ctxInfo = new StringBuilder();
      
         // Use ContextUtil to fetch context information
         ctxInfo.AppendFormat("Context ID:  {0}\n", ContextUtil.ContextId);
         ctxInfo.AppendFormat("Activity ID: {0}\n", ContextUtil.ActivityId);
        
         // If in transaction, get transaction ID
         string txId = "No Tx";
         if (ContextUtil.IsInTransaction)
            txId = ContextUtil.TransactionId.ToString();

         ctxInfo.AppendFormat("Transaction ID:    {0}\n", txId);
         ctxInfo.AppendFormat("Security Enabled?: {0}\n", 
            ContextUtil.IsSecurityEnabled);

         ctxInfo.Append("Stuff 18r");
         return ctxInfo.ToString();
      }
   }
}